platform/darwin/src/MGLMapCamera.mm

```diff
@@ -98,28 +98,26 @@ - (NSString *)description
             NSStringFromClass([self class]), (void *)self, _centerCoordinate.latitude, _centerCoordinate.longitude, _altitude, _heading, _pitch];
 }
 
-- (BOOL)isEqual:(id)other
+- (BOOL)isEqual:(id)otherCamera
 {
-    if ( ! [other isKindOfClass:[self class]])
-    {
-        return NO;
-    }
-    if (other == self)
-    {
-        return YES;
-    }
-    
-    MGLMapCamera *otherCamera = other;
-    return (_centerCoordinate.latitude == otherCamera.centerCoordinate.latitude
-            && _centerCoordinate.longitude == otherCamera.centerCoordinate.longitude
-            && _altitude == otherCamera.altitude
-            && _pitch == otherCamera.pitch && _heading == otherCamera.heading);
+    NSAssert([otherCamera isKindOfClass:[self class]], @"must compare two cameras");
```

Comment:
The method takes an id, which implies you can pass in any object. Do other classes assert when you attempt to compare to something of a different class?